query-engine/query-engine-node-api/src/log_callback.rs

```diff
@@ -29,6 +29,14 @@ pub struct LogCallback {
 impl LogCallback {
     pub fn new(env: Env, function: JsFunction) -> napi::Result<Self> {
         let name = env.create_string("prisma log callback")?;
+
+        // TODO: ThreadsafeFunctionState is not Send (because `napi_threadsafe_function` is not
+        // Send), so both the Mutex and the Arc are neither Send nor Sync, so it would still be a
+        // compile error to try to access `state` from multiple threads (in safe Rust). If
+        // `finalize_callback` can run on a different thread than the one that created `state`,
+        // this needs a comment explaining the control flow and safety. If it can't, it may be safe
+        // to replace this with `Rc<RefCell<ThreadsafeFunctionState>>`.
+        #[allow(clippy::arc_with_non_send_sync)]
```

Comment:
Note: this file was removed entirely in https://github.com/prisma/prisma-engines/pull/4168.

Reply:
Should that PR land first then?

Reply:
already landed, I'll rebase my commit out, thanks for the heads up!